### PR TITLE
Document KeyBinding.isPressed and KeyBinding.wasPressed

### PR DESCRIPTION
**Pull request by enbrain**
Questions:
* Should I mention MC-118107? https://bugs.mojang.com/browse/MC-118107
* Any better names for the methods?

### CODE REVIEW

File: mappings/net/minecraft/client/option/KeyBinding.mapping [hunk 1]
```diff
@@ -51,9 +51,32 @@ CLASS net/minecraft/class_304 net/minecraft/client/option/KeyBinding
 	METHOD method_1433 matchesMouse (I)Z
 		ARG 1 code
 	METHOD method_1434 isPressed ()Z
+		COMMENT Checks if the key is being held down.
```
Comment (liach) [inline mappings/net/minecraft/client/option/KeyBinding.mapping]:
```suggestion
		COMMENT {@return if the key is being held down}
```

File: mappings/net/minecraft/client/option/KeyBinding.mapping [hunk 2]
```diff
@@ -51,9 +51,32 @@ CLASS net/minecraft/class_304 net/minecraft/client/option/KeyBinding
...
+		COMMENT
+		COMMENT <p>Note that if you are continuously calling this method (like every
+		COMMENT tick), it doesn't always catch all key presses. This is because the key
+		COMMENT can be pressed and released before the next check.
+		COMMENT
+		COMMENT @see #wasPressed()
 	METHOD method_1435 equals (Lnet/minecraft/class_304;)Z
 		ARG 1 other
 	METHOD method_1436 wasPressed ()Z
+		COMMENT Checks if the key was pressed.
```
Comment (liach) [inline mappings/net/minecraft/client/option/KeyBinding.mapping]:
```suggestion
		COMMENT {@return if the key was pressed}
```

**Review by liach — COMMENTED** [review]
https://docs.oracle.com/en/java/javase/16/docs/specs/javadoc/doc-comment-spec.html#return

**Comment by haykam821** [comment]
I think we should introduce documentation features from newer JDKs in a later pull request.

**Review by haykam821 — APPROVED** [review]

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/client/option/KeyBinding.mapping [hunk 3]
```diff
@@ -51,9 +51,32 @@ CLASS net/minecraft/class_304 net/minecraft/client/option/KeyBinding
 	METHOD method_1433 matchesMouse (I)Z
 		ARG 1 code
 	METHOD method_1434 isPressed ()Z
+		COMMENT Checks if the key is being held down.
+		COMMENT
+		COMMENT <p>Note that if you are continuously calling this method (like every
+		COMMENT tick), it doesn't always catch all key presses. This is because the key
+		COMMENT can be pressed and released before the next check.
+		COMMENT
+		COMMENT @see #wasPressed()
```
Comment (liach) [inline mappings/net/minecraft/client/option/KeyBinding.mapping:60]:
```suggestion
		COMMENT @see #wasPressed()
		COMMENT @see <a href="https://bugs.mojang.com/browse/MC-118107">MC-118107</a>
```

File: mappings/net/minecraft/client/option/KeyBinding.mapping [hunk 4]
```diff
@@ -51,9 +51,32 @@ CLASS net/minecraft/class_304 net/minecraft/client/option/KeyBinding
...
+		COMMENT <p>To consume all remaining key presses, while-loop idiom can be used:
+		COMMENT <pre>
+		COMMENT {@code
+		COMMENT while(keyBinding.wasPressed()) {
+		COMMENT   // do your action
+		COMMENT }
+		COMMENT }
+		COMMENT </pre>
+		COMMENT
+		COMMENT @see #isPressed()
```
Comment (liach) [inline mappings/net/minecraft/client/option/KeyBinding.mapping:79]:
```suggestion
		COMMENT @see #isPressed()
		COMMENT @see <a href="https://bugs.mojang.com/browse/MC-118107">MC-118107</a>
```

**Review by liach — COMMENTED** [review]
In case you are concerned about the visibility of that issue.

**Pull request merged**